open-craft/sprints
# BB-3992: Change logic of estimated reviewer time on tasks

This changes the logic of review time estimation for tasks, according to the story points assignated to a task.
Here the estimations are defined on an environment variable

Ticket: https://gitlab.com/opencraft/dev/sprints/-/issues/2

Test Instructions:
- Check that unit tests makes sense(and cover correctly the implemented logic)
- I created some dummy tickets to test the review time estimation
-- Create and set an `.env` file (like [this](https://vault.opencraft.com:8200/ui/vault/secrets/secret/show/core/Sprints%20Development%20Envs)) and add the following environment variable
`SPRINT_HOURS_RESERVED_FOR_REVIEW={"null": 2, "1.9": 0.5, "2": 1, "3": 2, "5": 3, "5.1": 5}`
--Start the backend with `docker-compose -f local.yml up --build`
-- Run migrations with `docker-compose -f local.yml exec django python manage.py migrate`.(if you are running it for the first time)
-- Open the django shell with  `docker-compose -f local.yml exec django python manage.py shell_plus` to execute the following code snippets
-- Run this snippet so to create a Mocked/Fake Dashboard, to use its methods on our test
``` python
conn = CustomJira(
    server=settings.JIRA_SERVER,
    basic_auth=(settings.JIRA_USERNAME, settings.JIRA_PASSWORD),
    options={
        'agile_rest_path': GreenHopperResource.AGILE_BASE_REST_PATH,
        'headers': {
            'Referer': 'https://tasks.opencraft.com/download/resources/com.spartez.jira.plugins.jiraplanningpoker/frontend/index.html',
        }
    }
)

dashboard = Dashboard(26, conn)
```
-- Then run this snippet to prepare the required parameters for DashboardIssue class
``` python
quickfilters: List[QuickFilter] = conn.quickfilters(26)
members = get_cell_members(quickfilters)
sprints = get_all_sprints(conn, 26)
active_sprint_ids = {sprint.id for sprint in sprints['active']}
```
-- Run this to get the `dummy issues` we are going to use, along with the expected "review time" of each one
``` python
issues = [
    (conn.issue("BB-4660"), 2.0 * 3600),
    (conn.issue("BB-4661"), 0.5 * 3600),
    (conn.issue("BB-4662"), 1.0 * 3600),
    (conn.issue("BB-4663"), 2.0 * 3600),
    (conn.issue("BB-4664"), 3.0 * 3600),
    (conn.issue("BB-4665"), 5.0 * 3600),
    (conn.issue("BB-4666"), 3.0 * 3600)]
```
-- Finally run this to verify the method is behaving as expected for each dummy issue
``` python
for issue, expected_time_seconds in issues:
    dissue = DashboardIssue(issue, active_sprint_ids, members, dashboard.unassigned_user, dashboard.other_cell, issue_fields, dashboard.cell.key)
    assert dissue.review_time == expected_time_seconds
```
